### ISSUE TITLE
Failed to train llm and flow from scratch

### ISSUE THREAD

**Issue by zhangyike** [FunAudioLLM/CosyVoice#176]
Hi, I used about 3k hours mandarin speech data to train a llm and a flow model from scratch following the setting in cosyvoice.fromscratch.yaml. After training llm for 20 epochs and flow for 10 epochs, the models failed to synthesize any valid audio (usually a pure noise) neither in zeroshot mode nor in the SFT mode (specify a spk embeding in the training data).
When I replace the new trained llm  with the released one in CosyVoice-300M. The system can synthesize valid audio in SFT mode, but it still synthesizes noise in zeroshot mode. 
Additionally, it seems llm do not converge according to logs:

024-07-17 17:04:23,040 DEBUG TRAIN Batch 0/100 loss 2.799941 acc 0.000000 lr 0.00000272 grad_norm 0.000000 rank 0
2024-07-17 18:58:24,167 DEBUG TRAIN Batch 1/100 loss 1.066963 acc 0.266024 lr 0.00088160 grad_norm 0.000000 rank 0
2024-07-17 20:52:44,946 DEBUG TRAIN Batch 2/100 loss 1.127752 acc 0.242424 lr 0.00176152 grad_norm 0.000000 rank 0
2024-07-17 22:46:38,072 DEBUG TRAIN Batch 3/100 loss 1.004109 acc 0.286519 lr 0.00174165 grad_norm 0.000000 rank 0
2024-07-18 00:39:24,492 DEBUG TRAIN Batch 4/100 loss 0.909483 acc 0.322485 lr 0.00150840 grad_norm 0.000000 rank 0
2024-07-18 02:31:36,560 DEBUG TRAIN Batch 5/100 loss 0.774241 acc 0.408790 lr 0.00134928 grad_norm 0.000000 rank 0
2024-07-18 04:24:39,257 DEBUG TRAIN Batch 6/100 loss 1.018024 acc 0.270468 lr 0.00123164 grad_norm 0.000000 rank 0
2024-07-18 06:16:31,736 DEBUG TRAIN Batch 7/100 loss 0.956903 acc 0.317922 lr 0.00114029 grad_norm 0.000000 rank 0
2024-07-18 08:08:23,048 DEBUG TRAIN Batch 8/100 loss 0.773102 acc 0.376847 lr 0.00106692 grad_norm 0.000000 rank 0
2024-07-18 10:00:49,772 DEBUG TRAIN Batch 9/100 loss 0.971462 acc 0.275979 lr 0.00100611 grad_norm 0.000000 rank 0
2024-07-18 11:52:44,937 DEBUG TRAIN Batch 10/100 loss 0.946243 acc 0.305126 lr 0.00095440 grad_norm 0.000000 rank 0
2024-07-18 13:45:08,054 DEBUG TRAIN Batch 11/100 loss 0.970181 acc 0.286624 lr 0.00090986 grad_norm 0.000000 rank 0
2024-07-18 15:39:08,602 DEBUG TRAIN Batch 12/100 loss 0.927189 acc 0.326170 lr 0.00087108 grad_norm 0.000000 rank 0
2024-07-18 17:32:09,537 DEBUG TRAIN Batch 13/100 loss 0.886298 acc 0.353416 lr 0.00083689 grad_norm 0.000000 rank 0
2024-07-18 19:27:13,602 DEBUG TRAIN Batch 14/100 loss 0.851201 acc 0.354917 lr 0.00080643 grad_norm 0.000000 rank 0
2024-07-18 21:22:21,214 DEBUG TRAIN Batch 15/100 loss 0.864994 acc 0.361191 lr 0.00077922 grad_norm 0.000000 rank 0
2024-07-18 23:17:37,362 DEBUG TRAIN Batch 16/100 loss 0.927787 acc 0.323515 lr 0.00075442 grad_norm 0.000000 rank 0
2024-07-19 01:09:59,140 DEBUG TRAIN Batch 17/100 loss 0.884542 acc 0.341901 lr 0.00073199 grad_norm 0.000000 rank 0
2024-07-19 03:02:25,849 DEBUG TRAIN Batch 18/100 loss 0.890127 acc 0.338167 lr 0.00071134 grad_norm 0.000000 rank 0
2024-07-19 04:54:19,581 DEBUG TRAIN Batch 19/100 loss 0.852051 acc 0.346004 lr 0.00069233 grad_norm 0.000000 rank 0
2024-07-19 06:46:37,607 DEBUG TRAIN Batch 20/100 loss 1.011003 acc 0.262415 lr 0.00067486 grad_norm 0.000000 rank 0
2024-07-19 08:39:15,777 DEBUG TRAIN Batch 21/100 loss 1.065261 acc 0.260786 lr 0.00065863 grad_norm 0.000000 rank 0
2024-07-19 10:32:13,074 DEBUG TRAIN Batch 22/100 loss 0.881323 acc 0.339294 lr 0.00064349 grad_norm 0.000000 rank 0
2024-07-19 12:25:28,481 DEBUG TRAIN Batch 23/100 loss 0.998412 acc 0.283554 lr 0.00062932 grad_norm 0.000000 rank 0
2024-07-19 14:17:22,596 DEBUG TRAIN Batch 24/100 loss 0.923034 acc 0.322097 lr 0.00061611 grad_norm 0.000000 rank 0

and is seems the flow model converge well :
2024-07-17 18:53:52,767 DEBUG TRAIN Batch 0/100 loss 27.751081 lr 0.00000204 grad_norm 51.567730 rank 0
2024-07-17 22:15:13,573 DEBUG TRAIN Batch 1/100 loss 0.496736 lr 0.00042600 grad_norm 0.995903 rank 0
2024-07-18 01:36:04,590 DEBUG TRAIN Batch 2/100 loss 0.393265 lr 0.00085068 grad_norm 1.283468 rank 0
2024-07-18 04:56:39,048 DEBUG TRAIN Batch 3/100 loss 0.331865 lr 0.00088627 grad_norm 0.353341 rank 0
2024-07-18 08:18:40,431 DEBUG TRAIN Batch 4/100 loss 0.356549 lr 0.00076755 grad_norm 0.471500 rank 0
2024-07-18 11:40:14,633 DEBUG TRAIN Batch 5/100 loss 0.210071 lr 0.00068658 grad_norm 0.297273 rank 0
2024-07-18 15:02:51,103 DEBUG TRAIN Batch 6/100 loss 0.396542 lr 0.00062671 grad_norm 0.295585 rank 0
2024-07-18 18:21:29,930 DEBUG TRAIN Batch 7/100 loss 0.320616 lr 0.00058022 grad_norm 0.381744 rank 0
2024-07-18 21:41:30,448 DEBUG TRAIN Batch 8/100 loss 0.273874 lr 0.00054290 grad_norm 0.231301 rank 0
2024-07-19 01:02:13,137 DEBUG TRAIN Batch 9/100 loss 0.294604 lr 0.00051196 grad_norm 0.193230 rank 0
2024-07-19 04:22:40,842 DEBUG TRAIN Batch 10/100 loss 0.262893 lr 0.00048565 grad_norm 0.306482 rank 0
2024-07-19 07:43:59,952 DEBUG TRAIN Batch 11/100 loss 0.190467 lr 0.00046299 grad_norm 0.504046 rank 0
2024-07-19 11:05:02,095 DEBUG TRAIN Batch 12/100 loss 0.367979 lr 0.00044327 grad_norm 0.652572 rank 0
2024-07-19 14:24:47,537 DEBUG TRAIN Batch 13/100 loss 0.332409 lr 0.00042586 grad_norm 0.614597 rank 0

Anyone can help me?

**Comment by aluminumbox** [comment]
use cosyvoice.fromscratch.yaml if you want to train a small model on limited data

**Comment by zhangyike** [comment]
> cosyvoice.fromscratch.yaml.

I already used  cosyvoice.fromscratch.yaml.

**Comment by zhangyike** [comment]
I solved the problem according to  https://github.com/FunAudioLLM/CosyVoice/issues/158#issuecomment-2255377691

**Comment by taalua** [comment]
HI, how do you solve this issue? what code modification need to be done? Thanks

**Comment by vuminhnghiawork** [comment]
@zhangyike hi, can you share for me an example code to train cosyvoice from scratch please? thank you so much
